### PR TITLE
Timing benchmarks

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add a timing benchmark which is the echoBenchmark with the actual data transmission removed.  

### Proposed changes

- add timing benchmark
- add doc file describing the benchmarks
- rename the benchmarks to remove '_' from the first part of the name.

### CODE REVIEW

File: benchmarks/helics/ActionMessageBenchmarks.cpp [hunk 1]
```diff
@@ -39,9 +39,9 @@ static void BM_AM_FromString(benchmark::State& state)
     }
 }
 // Register the function as a benchmark
-BENCHMARK(BM_AM_FromString);
+BENCHMARK(BMfromString);
 
-static void BM_AM_toString_time(benchmark::State& state)
+static void BMtoString_time(benchmark::State& state)
```
Comment (nightlark) [inline benchmarks/helics/ActionMessageBenchmarks.cpp]: Should these be named `BMtoStringTime`?

File: docs/developer-guide/benchmarks.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# HELICS Benchmarks
+
+The HELICS repository has a few benchmarks that are intendended to test various aspects of the code and record performance over time
+
+## Baseline benchmarks
+
+### ActionMessage
+    microbenchmarks to test some operations concerning the serialization of the underlying message Structure in HELICS
```
Comment (nightlark) [inline docs/developer-guide/benchmarks.md]:
```suggestion
    microbenchmarks to test some operations concerning the serialization of the underlying message structure in HELICS
```

File: docs/developer-guide/benchmarks.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+### Conversion
+    microbenchmarks to test the serialization and deserialization of common data types in HELICS
+    
+## Simulation benchmarks
+
+### Echo
+    a set of federates representing a hub and spoke model of communication for value based interfaces
+    
+### Echo Message
+    a set of federates representing a hub and spoke model of communications for message based interfaces
```
Comment (nightlark) [inline docs/developer-guide/benchmarks.md]:
```suggestion
    a set of federates representing a hub and spoke model of communication for message based interfaces
```

File: docs/developer-guide/benchmarks.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    a variant of the Echo message test that add filters to the messages
+    
+### ring Benchmark 
+    a ring like structure that passes a value token around a bunch of times
+    
+### timing Benchmark
+    similar to echo but doesn't actually send any data just pure test of the timing messages
+    
+## Message Benchmarks
+
```
Comment (nightlark) [inline docs/developer-guide/benchmarks.md]:
```suggestion
```

File: benchmarks/helics/ActionMessageBenchmarks.cpp (hunk 1 above)
Comment (phlptp) [reply benchmarks/helics/ActionMessageBenchmarks.cpp]: yes, thanks!

**Pull request merged**